searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp

```diff
@@ -0,0 +1,623 @@
+}
+
+TEST_F(SlimeFillerTest, insert_tensor)
+{
+    TensorDataType   data_type(ValueType::from_spec("tensor(x{},y{})"));
+    TensorFieldValue value(data_type);
+    value = make_tensor(TensorSpec("tensor(x{},y{})").add({{"x", "4"}, {"y", "5"}}, 7));
+    expect_insert(make_slime_tensor_string(*value.getAsTensorPtr()), value);
+    expect_insert(R"("0x")", TensorFieldValue());
+}
+
+TEST_F(SlimeFillerTest, insert_reference)
+{
+    {
+        SCOPED_TRACE("normal reference");
+        ReferenceFieldValue value(get_as_ref_type("Reference<target_dummy_document>"),
+                                  DocumentId("id:ns:target_dummy_document::foo"));
+        expect_insert(R"("id:ns:target_dummy_document::foo")", value);
+    }
+    {
+        SCOPED_TRACE("empty reference");
+        ReferenceFieldValue value(get_as_ref_type("Reference<target_dummy_document>"));
+        expect_insert(R"("")", value);
+    }
+}
+
+TEST_F(SlimeFillerTest, insert_array)
+{
+    auto array = make_array();
+    expect_insert(R"(["foo","bar","baz"])", array);
+}
+
+TEST_F(SlimeFillerTest, insert_array_filtered)
+{
+    auto array = make_array();
+    expect_insert_filtered(R"(["foo","bar","baz"])", array, {0, 1, 2});
+    expect_insert_filtered("null", array, {});
+    expect_insert_filtered(R"(["foo"])", array, {0});
+    expect_insert_filtered(R"(["bar"])", array, {1});
+    expect_insert_filtered(R"(["baz"])", array, {2});
+    expect_insert_filtered("null", array, {0, 1, 2, 3});
+}
+
+TEST_F(SlimeFillerTest, insert_weighted_set)
+{
+    auto wset = make_weighted_set();
+    expect_insert(R"([{"item":"foo","weight":2},{"item":"bar","weight":4},{"item":"baz","weight":6}])", wset);
+}
+
+TEST_F(SlimeFillerTest, insert_weighted_set_filtered)
+{
+    auto wset = make_weighted_set();
+    expect_insert_filtered(R"([{"item":"foo","weight":2},{"item":"bar","weight":4},{"item":"baz","weight":6}])", wset, {0, 1, 2});
+    expect_insert_filtered("null", wset, {});
+    expect_insert_filtered(R"([{"item":"foo","weight":2}])", wset, {0});
+    expect_insert_filtered(R"([{"item":"bar","weight":4}])", wset, {1});
+    expect_insert_filtered(R"([{"item":"baz","weight":6}])", wset, {2});
+    expect_insert_filtered("null", wset, {0, 1, 2, 3});
+}
+
+TEST_F(SlimeFillerTest, insert_map)
+{
+    auto map = make_map();
+    expect_insert(R"([{"key":"key1","value":"value1"},{"key":"key2","value":"value2"},{"key":"key3","value":"value3"}])", map);
+}
+
+TEST_F(SlimeFillerTest, insert_map_filtered)
+{
+    auto map = make_map();
+    expect_insert_filtered(R"([{"key":"key1","value":"value1"},{"key":"key2","value":"value2"},{"key":"key3","value":"value3"}])", map, {0, 1, 2});
+    expect_insert_filtered("null", map, {});
+    expect_insert_filtered(R"([{"key":"key1","value":"value1"}])", map, {0});
+    expect_insert_filtered(R"([{"key":"key2","value":"value2"}])", map, {1});
+    expect_insert_filtered(R"([{"key":"key3","value":"value3"}])", map, {2});
```

Comment:
Consider adding a single test case for 2 elements as well.

Reply:
Done.